src/hipscat/catalog/pixel_tree.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import annotations
+
+import pandas as pd
+
+from hipscat.catalog import PixelNode, PixelNodeType
+
+
+class PixelTree:
+    """Sparse Quadtree of HEALPix pixels that make up the HiPSCat catalog
+
+    This class store each node in the tree, with leaf nodes corresponding to pixels with data files.
+
+    There are a number of methods in this class which allow for quickly navigating through the
+    tree and performing operations to filter the pixels in the catalog.
+
+    Attributes: pixels: Nested dictionary of pixel nodes stored in the tree. Indexed by HEALPix
+    order then pixel number root_pixel: Root node of the tree. It's children are a subset of the
+    12 base HEALPix pixels
+    """
+
+    METADATA_ORDER_COLUMN_NAME = "order"
+    METADATA_PIXEL_COLUMN_NAME = "pixel"
+
+    def __init__(self, partition_info_df: pd.DataFrame):
+        """Initialises the Tree from the partition info metadata
+
+        Args:
+            partition_info_df: Partition Info metadata file loaded into a pandas DataFrame
+
+        Raises:
+            ValueError: Incorrectly configured pixel structure in metadata file
+        """
+        self.root_pixel = PixelNode(-1, -1, PixelNodeType.ROOT, None)
+        self.pixels = {-1: {-1: self.root_pixel}}
+        self._create_tree(partition_info_df)
+
+    def __len__(self):
```

Comment:
To confirm I am reading this code correctly, it looks like it computes the number of nodes in the tree. Is that correct?

Regardless I think it is worth adding a comment to indicate exactly what length is being computed (number of leaves, number of nodes, number of children, etc.).

Reply:
Yeah, that's correct, and I'll add the docstring